vignettes/fitting.Rmd

```diff
@@ -354,9 +397,158 @@ dust2::dust_unfilter_run(unfilter, pars, adjoint = TRUE)
+
+```{r}
+dim(y)
+```
+
+representing
+
+* 4 state variables
+* 20 particles
+* 2 parameter groups
+* 101 times
+
+Consider just incidence as above:
+
+```{r}
+matplot(time, t(y[4, , 1, ]), type = "l", lty = 1, col = "#ff000055",
+        xlab = "Time (days)", ylab = "Cases", las = 1)
+matlines(time, t(y[4, , 2, ]), type = "l", lty = 1, col = "#0000ff55")
+points(cases ~ time, data, pch = 19)
+```
+
+## For dust2 filters/unfilters
+
+Here we assume (require, really) that each parameter set is associated with a different data set.  We may relax this in future, but this is the typical use case we have seen.  We need an additional column called `group` in addition to `time`:
+
+```{r, include = FALSE}
+data2 <- local({
+  d1 <- data
+  d2 <- data
+  d2$cases <- rpois(nrow(d2), d2$cases * 1.5)
+  data2 <- cbind(group = rep(1:2, each = nrow(d2)), rbind(d1, d2))
+  data2 <- data2[order(data2$time), ]
+  rownames(data2) <- NULL
+  data2
+})
+```
+
+```{r}
+head(data2)
+```
+
+(this is just synthetic data for now, created by duplicating and perturbing the original data).
+
+```
+plot(cases ~ time, data2, subset = group == 2, pch = 19, col = "red",
+     xlab = "Time (days)", ylab = "Cases")
+points(cases ~ time, data2, subset = group == 1, pch = 19, col = "blue")
+```
+
+Because the data is grouped, we don't need to tell `dust2::dust_filter_create()` that we have two groups, though you can pass `n_groups = 2` here if you want, which will validate that you really do have exactly two groups in the data:
+
+```{r}
+filter2 <- dust2::dust_filter_create(sir(), 0, data2, n_particles = 200)
+```
+
+When passing parameters into the filter, you now should mirror the format used in `dust2::dust_system_run()`; a list of lists:
+
+```{r}
+dust2::dust_filter_run(filter2, pars2)
+```
+
+We now have two likelihoods returned by the filter; one per group.
+
+For the deterministic unfilter the process is the same:
+
+```{r}
+unfilter2 <- dust2::dust_unfilter_create(sir(), 0, data2)
+dust2::dust_unfilter_run(unfilter2, pars2)
+```
+
+however, our gradient has picked up a dimension:
+
+```{r}
+dust2::dust_unfilter_last_gradient(unfilter2)
+```
+
+Here, the first **column** is the gradient of the first parameter set, and the first **row** is the gradient of `beta` over parameter sets.
+
+Compare with the single parameter case:
+
+```{r}
+dust2::dust_unfilter_run(unfilter, pars2[[1]])
+dust2::dust_unfilter_last_gradient(unfilter)
+```
+
+## For mcstate2 models
+
+Here, rather than a parameter vector you would pass in a matrix, where each column represents a parameter set:
+
+```{r}
+p <- cbind(c(0.2, 0.1), c(0.25, 0.1))
+```
+
+However, there's a small bit of functionality we need to add here in dust2, still.
+
+<!-- once https://github.com/mrc-ide/dust2/pull/59 is merged we can add: -->
+
+<!-- ```{r} -->
+<!-- likelihood2 <- dust2::dust_filter_mcstate(filter2, sir_packer) -->
+<!-- mcstate2::mcstate_model_density(likelihood2, p) -->
+<!-- ``` -->
```

Comment:
I've tried to add this as a suggestion, but the backticks mess each other up! Anyway - #59 is now merged, so we can include this